runtime/module.go

```diff
@@ -0,0 +1,122 @@
+package runtime
+
+import (
+	"fmt"
+
+	"github.com/gogo/protobuf/grpc"
+
+	"github.com/cosmos/cosmos-sdk/container"
+
+	runtimev1alpha1 "cosmossdk.io/api/cosmos/app/runtime/v1alpha1"
+	"cosmossdk.io/core/appmodule"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/std"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+)
+
+// BaseAppOption is a container.AutoGroupType which can be used to pass
+// BaseApp options into the container. It should be used carefully.
+type BaseAppOption func(*baseapp.BaseApp)
+
+// IsAutoGroupType indicates that this is a container.AutoGroupType.
+func (b BaseAppOption) IsAutoGroupType() {}
```

Comment:
Consider renaming `AutoGroupType` here, this type seems to have been renamed to [ManyPerContainerType](https://github.com/cosmos/cosmos-sdk/blob/27869a5af3f02aab0a8cc94ba3e36443b67e3824/container/group.go#L16) on `main`.